benchmark/benchmain/main.go

```diff
@@ -259,30 +270,39 @@ func init() {
 		runMode[0] = true
 		runMode[1] = true
 	default:
-		log.Fatalf("Unknown workloads setting: %v (want %v, %v, or %v)",
+		log.Fatalf("Unknown workloads setting: %v (want one of: %v)",
 			workloads, strings.Join(allWorkloads, ", "))
 	}
-	switch compressorMode {
-	case compressionOn:
-		enableCompressor = []bool{true}
-	case compressionOff:
-		enableCompressor = []bool{false}
-	case compressionBoth:
-		enableCompressor = []bool{false, true}
-	default:
-		log.Fatalf("Unknown compression mode setting: %v (want one of: %v)",
-			compressorMode, strings.Join(allCompressionModes, ", "))
-	}
-	if traceMode {
-		enableTrace = []bool{true}
-	}
+	enableCompressor = setMode(compressorMode)
+	enableTrace = setMode(traceMode)
 	// Time input formats as (time + unit).
 	readTimeFromInput(&ltc, readLatency)
 	readIntFromIntSlice(&kbps, readKbps)
 	readIntFromIntSlice(&mtu, readMtu)
 	readIntFromIntSlice(&maxConcurrentCalls, readMaxConcurrentCalls)
 	readIntFromIntSlice(&reqSizeBytes, readReqSizeBytes)
 	readIntFromIntSlice(&respSizeBytes, readRespSizeBytes)
+	// Re-write latency, kpbs and mtu if network mode is set.
+	if network, ok := networks[networkMode]; ok {
+		ltc = []time.Duration{network.Latency}
+		kbps = []int{network.Kbps}
+		mtu = []int{network.MTU}
+	}
+}
+
+func setMode(name string) []bool {
+	switch name {
+	case modeOn:
+		return []bool{true}
+	case modeOff:
+		return []bool{false}
+	case modeBoth:
+		return []bool{false, true}
+	default:
+		log.Fatalf("Unknown %s setting: %v (want one of: %v)",
+			name, name, strings.Join(allCompressionModes, ", "))
+	}
+	return []bool{}
```

Comment:
This should not be necessary; it is unreachable.

Reply:
I have to add a return either here of inside "default". Otherwise it has a "no return" error.

Reply:
Error from where?  I thought Go was smart enough to recognize Fatalf did not return.